app/antedecorators/fee/fixed_gas_test.go

```diff
@@ -0,0 +1,231 @@
+package fee_test
+
+import (
+	"testing"
+
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/simapp"
+	"github.com/cosmos/cosmos-sdk/testutil/testdata"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
+
+	"github.com/NibiruChain/nibiru/app"
+	"github.com/NibiruChain/nibiru/app/antedecorators/fee"
+	oracletypes "github.com/NibiruChain/nibiru/x/oracle/types"
+)
+
+func (suite *AnteTestSuite) TestOraclePostPriceTransactionsHaveFixedPrice() {
+	priv1, _, addr := testdata.KeyTestPubAddr()
+
+	tests := []struct {
+		name        string
+		messages    []sdk.Msg
+		expectedGas sdk.Gas
+		expectedErr error
+	}{
+		{
+			name: "Oracle Prevote Transaction",
+			messages: []sdk.Msg{
+				&oracletypes.MsgAggregateExchangeRatePrevote{
+					Hash:      "dummyData",
+					Feeder:    addr.String(),
+					Validator: addr.String(),
+				},
+			},
+			expectedGas: fee.OracleMessageGas,
+			expectedErr: nil,
+		},
+		{
+			name: "Oracle Vote Transaction",
+			messages: []sdk.Msg{
+				&oracletypes.MsgAggregateExchangeRateVote{
+					Salt:          "dummySalt",
+					ExchangeRates: "someData",
+					Feeder:        addr.String(),
+					Validator:     addr.String(),
+				},
+			},
+			expectedGas: fee.OracleMessageGas,
+			expectedErr: nil,
+		},
+		{
+			name: "Two messages in a transaction, one of them is an oracle vote message should fail (with MsgAggregateExchangeRatePrevote)",
+			messages: []sdk.Msg{
+				&oracletypes.MsgAggregateExchangeRatePrevote{
+					Hash:      "",
+					Feeder:    addr.String(),
+					Validator: addr.String(),
+				},
+				&types.MsgSend{
+					FromAddress: addr.String(),
+					ToAddress:   addr.String(),
+					Amount:      sdk.NewCoins(sdk.NewInt64Coin(app.BondDenom, 100)),
+				},
+			},
+			expectedGas: 0x151a,
```

Comment:
For the expected gas, can you use either the `strconv.FormatInt` of `fmt.Sprintf("%x", number)` to make the hexadecimal in the tests readable rather than including directing hexadecimal?

Reply:
absolutely